cfgmgr/portmgr.cpp

```diff
@@ -16,17 +17,57 @@ PortMgr::PortMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, c
         m_cfgPortTable(cfgDb, CFG_PORT_TABLE_NAME),
         m_cfgLagTable(cfgDb, CFG_LAG_TABLE_NAME),
         m_statePortTable(stateDb, STATE_PORT_TABLE_NAME),
-        m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME)
+        m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME),
+        m_appPortTable(appDb, APP_PORT_TABLE_NAME),
+        m_appLagTable(appDb, APP_LAG_TABLE_NAME)
 {
 }
 
-bool PortMgr::setPortMtu(const string &alias, const string &mtu)
+bool PortMgr::setPortMtu(const string &table, const string &alias, const string &mtu)
 {
     stringstream cmd;
     string res;
 
     cmd << IP_CMD << " link set dev " << alias << " mtu " << mtu;
-    return exec(cmd.str(), res) == 0;
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+    if (table == CFG_PORT_TABLE_NAME)
+    {
+        // Set the port MTU in application database to update both
+        // the port MTU and possibly the port based router interface MTU
+        vector<FieldValueTuple> fvs;
+        FieldValueTuple fv("mtu", mtu);
+        fvs.push_back(fv);
+        m_appPortTable.set(alias, fvs);
+    }
+    else if (table == CFG_LAG_TABLE_NAME)
+    {
+        // Set the port channel MTU in application database to update
+        // the LAG based router interface MTU in orchagent
+        vector<FieldValueTuple> fvs;
+        FieldValueTuple fv("mtu", mtu);
+        fvs.push_back(fv);
+        m_appLagTable.set(alias, fvs);
+
+        m_cfgLagTable.get(alias, fvs);
+        for (auto fv: fvs)
+        {
+            // Set the port channel members MTU in application database
+            // to update the port MTU in orchagent
+            if (fvField(fv) == "members")
```

Comment:
Just a suggestion, using Table::hset() and Table::hget could reduce the lines of code greatly.